web/pages/dashboard/[...slugs]/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { Page } from 'web/components/page'
+
+import { fromPropz, usePropz } from 'web/hooks/use-propz'
+import { dashboardPath, getDashboardBySlug } from 'web/lib/firebase/dashboards'
+import { Dashboard } from 'common/dashboard'
+import { Title } from 'web/components/title'
+import { Spacer } from 'web/components/layout/spacer'
+import { Content } from 'web/components/editor'
+import { UserLink } from 'web/components/user-page'
+import { getUser, User } from 'web/lib/firebase/users'
+import { ShareIcon } from '@heroicons/react/solid'
+import clsx from 'clsx'
+import { Button } from 'web/components/button'
+import { useState } from 'react'
+import { ShareDashboardModal } from 'web/components/share-dashboard-modal'
+import { Row } from 'web/components/layout/row'
+import { Col } from 'web/components/layout/col'
+import { ENV_CONFIG } from 'common/envs/constants'
+import Custom404 from 'web/pages/404'
+
+export const getStaticProps = fromPropz(getStaticPropz)
+export async function getStaticPropz(props: { params: { slugs: string[] } }) {
```

Comment:
On reflection, I think we don't need the `Propz` pattern for dashboards.

It exists so that we can accommodate private instances where only authenticated users can load the data client-side.

But dashboards are an inherently public feature, so it's probably unnecessary and mucks up the code. If we really needed it private, we could always add it back. Plus, I think nowadays we would just use server side rendering to check your auth credentials first instead of using this pattern.

So, you can just turn this into `function getStaticProps` and don't call `usePropz` in the page component.